service_common.go

```diff
@@ -272,6 +271,7 @@ func (c *controller) addServiceBinding(svcName, svcID, nID, eID, containerName s
 			vip:      vip,
 			fwMark:   fwMarkCtr,
 			backEnds: make(map[string]net.IP),
```

Comment:
That approach would certainly work as well.  It would make some things cleaner and others a bit more awkward.  I'll give it a quick shot and see how it looks.